core/state_processor.go

```diff
@@ -170,6 +170,10 @@ func (p *StateProcessor) Process(block *types.Block, statedb *state.StateDB, cfg
 		ProcessConsolidationQueue(&requests, evm)
 	}
 
+	if err = checkNanoBlackList(statedb, evm); err != nil {
```

Comment:
Could we add this check in tx execution? It will fail when fill transaction in mining.